app/src/main/java/es/wolfi/app/passman/CredentialItemFragment.java

```diff
@@ -80,15 +86,41 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View view = inflater.inflate(R.layout.fragment_credential_item_list, container, false);
 
         // Set the adapter
-        if (view instanceof RecyclerView) {
-            Context context = view.getContext();
-            RecyclerView recyclerView = (RecyclerView) view;
+        View credentialView = view.findViewById(R.id.list);
+        if (credentialView instanceof RecyclerView) {
+            Context context = credentialView.getContext();
+            final RecyclerView recyclerView = (RecyclerView) credentialView;
             if (mColumnCount <= 1) {
                 recyclerView.setLayoutManager(new LinearLayoutManager(context));
             } else {
                 recyclerView.setLayoutManager(new GridLayoutManager(context, mColumnCount));
             }
-            Vault v = (Vault) SingleTon.getTon().getExtra(SettingValues.ACTIVE_VAULT.toString());
+            final Vault v = (Vault) SingleTon.getTon().getExtra(SettingValues.ACTIVE_VAULT.toString());
+            final EditText searchInput = (EditText) view.findViewById(R.id.search_input);
+            searchInput.addTextChangedListener(new TextWatcher() {
+                @Override
+                public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+                }
+
+                @Override
+                public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+                    ArrayList<Credential> filteredCredentials = new ArrayList<Credential>(v.getCredentials());
```

Comment:
Maybe we should have this functionality more reusable and on a callable asynchronous container, with long vaults with hundred of passwords it might lead to ANR on slow devices, don't you think

Reply:
It's definitely not very efficient.

I played around with the idea of having the search/filtering triggered after pressing a button rather than onTextChanged but the effect wasn't as nice. An asynchronous call might be a nice compromise.

Reply:
I would appreciate if you try to make the search more reusable too, so i can use it for vaults and other items within the app.

Also note that translations are managed by transiflex, so any changes you make will be overwritten by the transiflex robot.

Reply:
I was wondering if you plan to keep going on this feature and polish it to the end.